Dockerfile.debug

```diff
@@ -140,13 +141,13 @@ EXPOSE 2020
 # Default: Run under valgrind
 # alternatively, you can change this to run FB directly and try
 # to get a core file instead
-CMD valgrind --leak-check=full --error-limit=no /fluent-bit/bin/fluent-bit -c /fluent-bit/etc/fluent-bit.conf
+# CMD valgrind --leak-check=full --error-limit=no /fluent-bit/bin/fluent-bit -c /fluent-bit/etc/fluent-bit.conf
 
 # Alternate: Run directly and get core file
 # make sure you run with `docker run --ulimit core=-1`
 # and mount the /cores folder onto the host
 # Un-Comment the lines below:
 # 
-# RUN mkdir /cores && chmod 777 /cores
-# CMD sysctl -w kernel.core_pattern='/cores/core.%e.%p' && /fluent-bit/bin/fluent-bit -c /fluent-bit/etc/fluent-bit.conf
+RUN mkdir /cores && chmod 777 /cores
+CMD /fluent-bit/bin/fluent-bit -c /fluent-bit/etc/fluent-bit.conf
```

Comment:
Actually, good comment. Should the normal debug build still be the valgrind version because customers should use that unless they really are struggling.

Reply:
Yea that was my thinking. Valgrind is the default since its easiest, this core dump option is more advanced. 

Though may be we should just have 3 Dockerfiles, one for prod, one for valgrind, and one for core... its more effort to maintain though.

Reply:
Maybe the easiest way (for maintenance) would be to modify the docker file to switch between the entry points based on some env variable that is passed in from the make file.

Reply:
Also the env variable would toggle getting gdb installed and the core folders being created.

Reply:
We can do this later

Reply:
I've tried to do things like this before and actually customizing dockerfiles is very not fun... an example of something similar to this is how we can build our go plugins from a fork